finality-aleph/src/party/manager/task.rs

```diff
@@ -12,26 +12,43 @@ pub type Handle = Pin<Box<(dyn Future<Output = sc_service::Result<(), ()>> + Sen
 pub struct Task {
     handle: Handle,
     exit: oneshot::Sender<()>,
+    error_on_exit: Option<bool>,
 }
 
 impl Task {
     /// Create a new task.
     pub fn new(handle: Handle, exit: oneshot::Sender<()>) -> Self {
-        Task { handle, exit }
+        Task {
+            handle,
+            exit,
+            error_on_exit: None,
+        }
     }
 
     /// Cleanly stop the task.
-    pub async fn stop(self) {
+    pub async fn stop(self) -> Result<(), ()> {
+        if let Some(res) = self.error_on_exit {
+            return if res { Err(()) } else { Ok(()) };
+        }
```

Comment:
Suggestion: Perhaps this would be clearer as:

```suggestion
        match self.error_on_exit {
            Some(true) => return Err(()),
            Some(false) => return Ok(()),
            None => ()
        }
```

This will also happen naturally if you go with my enum suggestion.